bundles/org.eclipse.cdt.lsp.editor.ui/src/org/eclipse/cdt/lsp/editor/ui/preference/LspEditorPreferencesTester.java

```diff
@@ -58,6 +60,17 @@ public boolean test(Object receiver, String property, Object[] args, Object expe
 			}
 			// When resource == null it's an external file: Check if the file is already opened, if not check the active editor:
 			return isFileOpenedInLspEditor(editorInput);
+		} else if (receiver instanceof ITranslationUnit) {
+			// called to enable the LS based CSymbolsContentProvider: 
+			var cProject = ((ITranslationUnit) receiver).getCProject();
```

Comment:
What do you think about something like
```java
Optional.of((ITranslationUnit) receiver)//
    .map(ITranslationUnit::getCProject)//
    .map(ICProject::getProject)//
    .ifPresent(this::preferLspEditor);

```

Reply:
Hm, I don't think the lambda would be a improvement here, since I have to return the result of `preferLspEditor`

Reply:
You are right, my snippet will not match as is
```java
return Optional.of((ITranslationUnit) receiver)//
 .map(ITranslationUnit::getCProject)//
 .map(ICProject::getProject)//
 .map(this::preferLspEditor)//
 .orElse(Boolean.FALSE);